platforms/access-bridge/src/controllers/access-controller.ts

```diff
@@ -0,0 +1,109 @@
+import { IncomingMessage, ServerResponse } from 'http';
+
+import { AccessControlPlan } from '@jwp/ott-common/types/plans.js';
+import { Viewer } from '@jwp/ott-common/types/access.js';
+
+import { ParameterInvalidError, AccessBridgeError, sendErrors, UnauthorizedError } from '../errors.js';
+import { AccessService } from '../services/access-service.js';
+import { PlansService } from '../services/plans-service.js';
+import { isValidSiteId, parseJsonBody } from '../utils.js';
+import { AccountService } from '../services/account-service.js';
+
+/**
+ * Controller class responsible for handling access-related services.
+ */
+export class AccessController {
+  private accountService: AccountService;
+  private accessService: AccessService;
+  private plansService: PlansService;
+
+  constructor() {
+    this.accountService = new AccountService();
+    this.accessService = new AccessService();
+    this.plansService = new PlansService();
+  }
+
+  /**
+   * Service handler for generating passport access tokens.
+   * @param req The HTTP request object.
+   * @param res The HTTP response object.
+   * @param params The request parameters containing site_id.
+   */
+  generatePassport = async (req: IncomingMessage, res: ServerResponse, params: { [key: string]: string }) => {
```

Comment:
Is there a reason that params isn't typed?

Reply:
updated with express.